### PR TITLE
Auto Update Bug Fixes

### PR DESCRIPTION
**Pull request by niteskum**
This PR include Fix for Issue #14266  and #14232 #14234

### CODE REVIEW

**Comment by niteskum** [comment]
ping @swmitra @navch @mbhavi  for review.

To track network connectivity failure while download is in progress , assigned a timeout value to download Request in **downloadInstaller** function in AutoUpdateDomain.js.
Currently I have kept timeout value is 30 seconds. If Socket Connection is idle for more than 30 seconds. Download Request will be aborted and Download Error Message "**Network is Disconnected or too slow.**" will be shown to user.

@swmitra  please suggest the timeout value. is timeout of 30 secs good or i should change it?

File: src/extensions/default/AutoUpdate/main.js [hunk 1]
```diff
@@ -624,6 +626,15 @@ define(function (require, exports, module) {
         if (checkIfOnline()) {
             postMessageToNode(MessageIds.PERFORM_CLEANUP, ['.json'], true);
         } else {
+            enableCheckForUpdateEntry(true);
+            UpdateStatus.cleanUpdateStatus();
+            HealthLogger.sendAnalyticsData(
```
Comment (shubhsnov) [inline src/extensions/default/AutoUpdate/main.js]: I also see analytics related stuff here. Shouldn't this be in a separate PR??

File: src/extensions/default/AutoUpdate/node/AutoUpdateDomain.js [hunk 2]
```diff
@@ -260,8 +261,10 @@
         updateParams = updateParams || _updateParams;
         try {
             var ext = path.extname(updateParams.installerName);
-            var localInstallerPath = path.resolve(updateDir, Date.now().toString() + ext);
-            progress(request(updateParams.downloadURL), {})
+            var localInstallerPath = path.resolve(updateDir, Date.now().toString() + ext),
+                localInstalllerFile = fs.createWriteStream(localInstallerPath),
+                requestCompleted = true;
+            progress(request(updateParams.downloadURL, {timeout: 180000}), {})
```
Comment (shubhsnov) [inline src/extensions/default/AutoUpdate/node/AutoUpdateDomain.js]: Suggestion: Move the timeout value to name variable for better readability.

File: src/extensions/default/AutoUpdate/main.js (hunk 1 above)
Comment (niteskum) [reply src/extensions/default/AutoUpdate/main.js]: No, Here It is just a call to api to send Analytics Data. No Framework level change.

**Review by shubhsnov — APPROVED** [review]

File: src/extensions/default/AutoUpdate/main.js [hunk 3]
```diff
@@ -925,6 +924,9 @@ define(function (require, exports, module) {
             if (message === _nodeErrorMessages.DOWNLOAD_ERROR) {
                 descriptionMessage = Strings.DOWNLOAD_ERROR;
             }
+            if (message === _nodeErrorMessages.NETWORK_SLOW_OR_DISCONNECTED) {
```
Comment (shubhsnov) [inline src/extensions/default/AutoUpdate/main.js]: Else if?

**Review by boopeshmahendran — APPROVED** [review]
